flang/lib/Semantics/resolve-directives.cpp

```diff
@@ -2112,7 +2112,7 @@ void OmpAttributeVisitor::Post(const parser::OpenMPAllocatorsConstruct &x) {
 
 static bool IsPrivatizable(const Symbol *sym) {
   auto *misc{sym->detailsIf<MiscDetails>()};
-  return !IsProcedure(*sym) && !IsNamedConstant(*sym) &&
+  return IsVariableName(*sym) && !IsProcedure(*sym) && !IsNamedConstant(*sym) &&
```

Comment:
This whole check can probably be simplified further in a separate patch.

Reply:
Right, it should be OK to remove at least `!IsNamedConstant(*sym)`, probably `!IsProcedure(*sym)` too.

Reply:
I will update in a separate patch.